File: LeetCode/1658. Minimum Operations to Reduce X to Zero/solution.py
class Solution:
    def minOperations(self, nums: List[int], x: int) -> int:
        n = len(nums)
        x = sum(nums) - x
        mp = {0: -1}
        cur, res = 0, -float('inf')
        for i, a in enumerate(nums):
            cur += a
            mp[cur] = i 
            if cur - x in mp:
                res = max(res, i-mp[cur-x])
        res = n - res
        return res if res < float('inf') else -1

class Solution:
    def minOperations(self, nums: List[int], x: int) -> int:
        n = len(nums)
        left, right = 0, 0
        cur, res = sum(nums), float('inf')
        for right in range(n+1):
            while cur < x and left < right:
                cur += nums[left]
                left += 1
            if cur == x:
                res = min(res,n-(right-left))
            if right == n:
                break
            cur -= nums[right]
        return res if res < float('inf') else -1

File: LeetCode/1658. Minimum Operations to Reduce X to Zero/solution.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int x) {
        int n = nums.size();
        int sums = 0, cur = 0;
        for (auto num : nums)
            sums += num;
        int y = sums - x;
        if (y < 0)
            return -1;
        if (y == 0)
            return n;
        int left = 0, res = -1;
        for (int right = 0; right < n; right++){
            cur += nums[right];
            while (cur > y)
                cur -= nums[left++];
            if (cur == y)
                res = max(res, right-left+1);
        }
        return res == -1 ? -1 : n-res;

    }
};